### DISCUSSION TITLE
How to ignore unregistered namespaces in i18next (colons in strings)

### DISCUSSION THREAD

**Discussion by abea** [i18next/i18next#1718] [Q&A]
We use i18next in a CMS to power its internationalization features. Since developers can build however they want with the CMS there is opportunity for them to add l10n keys that include colons, including as part of HTML, such as `Find more info <a href="https://info.net">here</a>`.

[As has been documented](https://github.com/i18next/i18next/issues/1112#issuecomment-429231561), with default namespace separator settings i18next will think the colon is identifying a namespace/key pair. Since the CMS uses its own namespace (so devs won't accidentally overwrite UI strings), we don't have the option to turn off namespacing completely (with `nsSeparator: false`).

What I'm looking for is a way for i18next to **only recognize registered namespaces as namespaces**. So if we tell i18next that the valid namespaces are `['ns1', 'ns2']` and it receives `Title: Subtitle`, that string will be treated as a key, not a namespace/key pair.

I saw the `loadNamespaces` method, but that looks to simply register them to the `ns` option on the instance. Is there a way for i18next to essentially disallow any unregistered namespace?

**Comment by jamuhl** [comment]
why not just chaning the nsSeparator to some value you won't have in your ns:key combination -> eg. `nsSeparator: '###'`

**Reply by abea** [reply]
The short answer right now is that it would be a BC break in this major version. We can't really expose that option for devs to change since the UI uses the `:` separator already (not easily, at least). It would also be a more confusing dev experience.

It seems the answer to my question is that there isn't such a feature right now, so we'll figure out some workaround. Thank you for answering.
